tests/unit/test_charm.py

```diff
@@ -818,6 +820,138 @@ def test_on_config_changed_with_ingress(
     assert app_data[0]["redirect_uri"].startswith(expected_redirect_url)
 
 
+def test_on_client_config_relation_removed_with_ingress(
+    harness: Harness,
+    mocked_container: Container,
+    mocked_migration_is_needed: MagicMock,
+    mocked_kratos_configmap: MagicMock,
+) -> None:
+    setup_peer_relation(harness)
+    setup_postgres_relation(harness)
+    setup_ingress_relation(harness, "public")
+    (_, login_databag) = setup_login_ui_relation(harness)
+
+    relation_id, data = setup_external_provider_relation(harness)
+    harness.remove_relation(relation_id)
+    container = harness.model.unit.get_container(CONTAINER_NAME)
+    harness.charm.on.leader_elected.emit()
```

Comment:
no need to call this event as harness triggers it anyway when the leader is set (see https://github.com/canonical/kratos-operator/blob/55ba14cdb9d71220445a6ba037e16c0bef3790f8/tests/unit/conftest.py#L23)

Reply:
There are some tests that set the leader to `False` (eg https://github.com/canonical/kratos-operator/blob/main/tests/unit/test_charm.py#L572) and the state persists. If I remove this the test fails.

This probably means that we should refactor our tests, but this is not the right time